### PR TITLE
Add EventCallback content to Components topic+sample

### LINKED ISSUE #11243: Document EventCallback
Under "Component parameters" (Data binding), I don't see an example on how the parent component get its variable updated when it is changed IN the child component. The example shows the change being made in the parent component, all the child component do is showing its own variable, and that works without "bind".

---
#### Dokumentinformasjon

⚠ *Ikke rediger denne delen. Den kreves for koblingen docs.microsoft.com ➟ GitHub-problem.*

* ID: 11a64802-62dc-40e0-034f-fcee55f31d4e
* Version Independent ID: 038d7f90-aa71-a35c-cee9-c8543403cfd7
* Content: [Create and use Razor Components](https://docs.microsoft.com/nb-no/aspnet/core/razor-components/components?view=aspnetcore-3.0#data-binding)
* Content Source: [aspnetcore/razor-components/components.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/razor-components/components.md)
* Product: **aspnet-core**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #11243 

[Internal Review Topic (EventCallback section)](https://review.docs.microsoft.com/en-us/aspnet/core/razor-components/components?view=aspnetcore-3.0&branch=pr-en-us-11924#eventcallback)

* For now, I focus on basic implementation. Let's add more detail where you think it's required from here.
* Interesting bits are in the *EventCallback* section (a subsection of the *Event handling* section) and the accompanying sample updates for that content, which run lines 325-371 of *aspnetcore/razor-components/components.md*. The section is linked via the *Internal Review Topic* link. The rest of the updates here are general updates, and we can review those bits on the docs side.
* **_Layouts!_** ... is there any way to use `EventCallback`/`EventCallback<T>` across a the component-layout barrier (i.e., affect the layout's objects from a component using the layout)?

### CODE REVIEW

**Comment by guardrex** [comment]
@rynowak Do u have time to look at the section on `EventCallback`?

If so, use the link ... it goes right to the important bits (the docs team can review the rest of this PR):

[Internal Review Topic (EventCallback section)](https://review.docs.microsoft.com/en-us/aspnet/core/razor-components/components?view=aspnetcore-3.0&branch=pr-en-us-11924#eventcallback)

This one should get u to the diff for line comments (lines 325-371):

https://github.com/aspnet/AspNetCore.Docs/pull/11924/files#diff-68aec1e5867e7fd4666f410fdf9b3a88

File: aspnetcore/razor-components/components.md [hunk 1]
```diff
@@ -322,6 +322,53 @@ It's often convenient to close over additional values, such as when iterating ov
...
+
+Invoke an `EventCallback` or `EventCallback<T>` with `InvokeAsync` and await the <xref:System.Threading.Tasks.Task>:
+
+```csharp
+await callback.InvokeAsync(arg);
+```
+
+Use `EventCallback` and `EventCallback<T>` for event handling and binding component parameters. Don't use `EventCallback` and `EventCallback<T>` for child content&mdash;continue to use `RenderFragment` and `RenderFragment<T>` for child content.
+
+Prefer the strongly typed `EventCallback<T>`, which provides better error feedback to users of the component. Use `EventCallback` when there's no value passed to the callback.
```
Comment (danroth27) [inline aspnetcore/razor-components/components.md]: I believe another benefit of EventCallback is that specifying the event parameter in your event callback is optional, just like it is for other UI event handlers.

File: aspnetcore/razor-components/components.md [hunk 2]
```diff
@@ -322,6 +322,53 @@ It's often convenient to close over additional values, such as when iterating ov
...
+
+[!code-cshtml[](common/samples/3.x/BlazorSample/Pages/ChildComponent.cshtml?highlight=5-7,17-18)]
+
+The Parent component sets the child's `EventCallback<T>` to its `ShowMessageTriggeredByChild` method.
+
+*Parent component*:
+
+[!code-cshtml[](common/samples/3.x/BlazorSample/Pages/ParentComponent.cshtml?name=snippet_ParentComponent&highlight=6,16-19)]
+
+When the button is selected in the Child component:
```
[the remark this answers is not in the document]
Comment (guardrex) [reply aspnetcore/razor-components/components.md]: MS style manual calls for "selected" over "clicked." It's ok tho to write "right-click(ed)" when needed.

**Comment by guardrex** [comment]
@rynowak It is based on your remarks. I made it a little plainer on the latest commit.

@danroth27 It may require another touch (or *two*), but it seems to be heading in the right direction.

It's ready for another :eye:.

**Pull request merged**